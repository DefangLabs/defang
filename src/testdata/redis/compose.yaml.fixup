{
  "no-ext": {
    "command": null,
    "entrypoint": null,
    "image": "redis",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6379,
        "protocol": "tcp"
      }
    ]
  },
  "no-ports": {
    "command": null,
    "entrypoint": null,
    "image": "redis",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6379,
        "protocol": "tcp"
      }
    ]
  },
  "no-ports-override": {
    "command": [
      "--port",
      "6380"
    ],
    "entrypoint": null,
    "image": "redis",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6380,
        "protocol": "tcp"
      }
    ]
  },
  "short-ports": {
    "command": null,
    "entrypoint": null,
    "image": "redis",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6379,
        "published": "6379",
        "protocol": "tcp"
      }
    ]
  },
  "with-ext": {
    "command": null,
    "entrypoint": null,
    "image": "redis",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6379,
        "protocol": "tcp"
      }
    ]
  },
  "wrong-image": {
    "command": null,
    "entrypoint": null,
    "image": "example",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6379,
        "protocol": "tcp"
      }
    ]
  }
}