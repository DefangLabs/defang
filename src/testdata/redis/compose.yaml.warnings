 ! service "no-ext": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "no-ext": stateful service will lose data on restart; use a managed service instead
 ! service "no-ports": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "with-ext": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "wrong-image": managed Redis service should use a redis image
 ! service "wrong-image": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
service "no-ports": managed Redis service should expose a host port