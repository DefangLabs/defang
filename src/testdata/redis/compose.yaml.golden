name: redis
services:
  no-ext:
    image: redis
    networks:
      default: null
    ports:
      - mode: host
        target: 6379
        protocol: tcp
  no-ports:
    image: redis
    networks:
      default: null
    x-defang-redis: null
  no-ports-override:
    command:
      - --port
      - "6380"
    image: redis
    networks:
      default: null
    x-defang-redis: null
  with-ext:
    image: redis
    networks:
      default: null
    ports:
      - mode: host
        target: 6379
        protocol: tcp
    x-defang-redis:
      allow-downtime: true
  wrong-image:
    image: example
    networks:
      default: null
    ports:
      - mode: host
        target: 6379
        protocol: tcp
    x-defang-redis: null
networks:
  default:
    name: redis_default
